scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala

```diff
@@ -86,8 +91,12 @@ private[scio] object ScioUtil {
     }
   }
 
-  def pathWithShards(path: String): String =
-    path.replaceAll("\\/+$", "") + "/part"
+  def pathWithPrefix(path: String, prefix: String = null): String = {
+    val replacement = s"/${Option(prefix).getOrElse("part")}"
+    // replaceAll surprisingly does the wrong thing with regex /*$
+    val stripped = StringUtils.stripEnd(path, "/")
+    s"${stripped}${replacement}"
```

Comment:
This part misses some cleaning
```suggestion
    val filePrefix = Option(prefix).getOrElse("part")
    val stripped = StringUtils.stripEnd(path, "/")
    s"${stripped}/${filePrefix}"
```

Reply:
Don't these do the same thing?

Reply:
I'm more focusing on the naming: 
- `replacement` -> there isn't any replacement
- `// replaceAll surprisingly does the wrong thing with regex /*$` we do not call `replaceAll`